transformer_engine/pytorch/module.py

```diff
@@ -417,23 +417,22 @@ def prepare_forward(
             self.fp8_init(num_gemms=num_gemms)
             self.set_fp8_weights()
 
-            # Either we're in FP8 training or calibration for FP8 inference
-            needs_stats = (self.training if self.fp8 else self.fp8_calibration)
-
-            if needs_stats:
-                update_weight_scale_inv = is_first_microbatch is None or is_first_microbatch
-                # Previous iteration was grad_enabled
-                if self.fp8_meta.get("update_amax_and_scale_fwd", False):
-                    if self.fp8_meta["recipe"].reduce_amax:
-                        copy_amax_from_global_buffer(self.fp8_meta, forward=True)
-                        amax_and_scale_update(
-                            self.fp8_meta, True, update_weight_scale_inv=update_weight_scale_inv
-                        )
-                        set_amax_buffer_key_deletion(self.fp8_meta, forward=True)
-                    else:
-                        amax_and_scale_update(
-                             self.fp8_meta, True, update_weight_scale_inv=update_weight_scale_inv
-                        )
+            update_weight_scale_inv = is_first_microbatch is None or is_first_microbatch
+
+            # Previous iteration was grad_enabled
+            if self.fp8_meta.get("update_amax_and_scale_fwd", False):
```

Comment:
Is fp8_meta always guaranteed to exist here?

Reply:
Yes, we initialize it in `TransformerEngineBaseModule`.